### PR TITLE
refactor(AboutModalBox): refactor CSS, redefine vars instead of redefining properties

### PR DESCRIPTION
**Pull request by mcoker**
fixes https://github.com/patternfly/patternfly-next/issues/980

### CODE REVIEW

File: src/patternfly/components/AboutModalBox/about-modal-box.scss [hunk 1]
```diff
@@ -235,19 +225,21 @@
...
+    --pf-c-about-modal-box__close--PaddingRight: var(--pf-c-about-modal-box__close--lg--PaddingRight);
+
     justify-content: flex-end; // align flex end the close over the narrow hero image
-    padding-right: var(--pf-c-about-modal-box__close--PaddingRight);
   }
 
   // Close button
   // Since this btn is unique to the about modal, use contextual selectors in this specific instance
-  .pf-c-button.pf-m-plain {
+  .pf-c-button {
```
Comment (christiemolloy) [inline src/patternfly/components/AboutModalBox/about-modal-box.scss]: This seems to be overriding a lot of properties of the button component, wondering if this is an opportunity to use `pf-c-about-modal-box__button-close`

File: src/patternfly/components/AboutModalBox/about-modal-box.scss [hunk 2]
```diff
@@ -235,19 +225,21 @@
...
     display: flex;
     align-content: center;
     justify-content: center;
```
Comment (christiemolloy) [inline src/patternfly/components/AboutModalBox/about-modal-box.scss]: can you assign the border-radius to a global variable ?

File: src/patternfly/components/AboutModalBox/about-modal-box.scss [hunk 3]
```diff
@@ -10,79 +10,72 @@
...
   // Logo
   // Set default padding right, bottom, left to shared values
-  --pf-c-about-modal-box__logo--PaddingRight: var(--pf-c-about-modal-box--PaddingRight);
-  --pf-c-about-modal-box__logo--PaddingBottom: var(--pf-c-about-modal-box--PaddingBottom);
-  --pf-c-about-modal-box__logo--PaddingLeft: var(--pf-c-about-modal-box--PaddingLeft);
+  --pf-c-about-modal-box__logo--PaddingRight: var(--pf-c-about-modal-box--Padding);
+  --pf-c-about-modal-box__logo--PaddingBottom: var(--pf-c-about-modal-box--Padding);
+  --pf-c-about-modal-box__logo--PaddingLeft: var(--pf-c-about-modal-box--Padding);
 
   // Logo image
```
Comment (christiemolloy) [inline src/patternfly/components/AboutModalBox/about-modal-box.scss]: Can you condense this to one assignment? `--pf-c-about-modal-box__logo-image--Padding`

Comment (christiemolloy) [inline src/patternfly/components/AboutModalBox/about-modal-box.scss]: For: 
`
--pf-c-about-modal-box__button-close--BackgroundColor: #{$pf-color-black-1000};
--pf-c-about-modal-box__button-close--hover--BackgroundColor: rgba($pf-color-black-1000, .4);
`

File: src/patternfly/components/AboutModalBox/about-modal-box.scss [hunk 4]
```diff
@@ -10,79 +10,72 @@
...
 
   // This is the min allowable height of the content
   --pf-c-about-modal-box__content--MinHeight: calc(1rem * 4 * var(--pf-global--LineHeight--md));
 
   // Logo
   // Set default padding right, bottom, left to shared values
-  --pf-c-about-modal-box__logo--PaddingRight: var(--pf-c-about-modal-box--PaddingRight);
-  --pf-c-about-modal-box__logo--PaddingBottom: var(--pf-c-about-modal-box--PaddingBottom);
-  --pf-c-about-modal-box__logo--PaddingLeft: var(--pf-c-about-modal-box--PaddingLeft);
+  --pf-c-about-modal-box__logo--PaddingRight: var(--pf-c-about-modal-box--Padding);
```
Comment (christiemolloy) [inline src/patternfly/components/AboutModalBox/about-modal-box.scss]: should you do `--pf-c-about-modal-box__logo--PaddingX` to condense Left and Right padding?

File: src/patternfly/components/AboutModalBox/about-modal-box.scss [hunk 5]
```diff
@@ -10,79 +10,72 @@
...
   --pf-c-about-modal-box__header-strapline--PaddingBottom: var(--pf-global--spacer--md);
 
   // Content
   // Using global padding values for margins to prevent scrollbar extending past content
   // Set default margin top, right, bottom, left to shared values
-  --pf-c-about-modal-box__content--MarginTop: var(--pf-c-about-modal-box--PaddingTop);
-  --pf-c-about-modal-box__content--MarginRight: var(--pf-c-about-modal-box--PaddingRight);
-  --pf-c-about-modal-box__content--MarginBottom: var(--pf-c-about-modal-box--PaddingBottom);
-  --pf-c-about-modal-box__content--MarginLeft: var(--pf-c-about-modal-box--PaddingLeft);
+  --pf-c-about-modal-box__content--Margin: var(--pf-c-about-modal-box--Padding);
```
Comment (christiemolloy) [inline src/patternfly/components/AboutModalBox/about-modal-box.scss]: Is using Padding for Margin an exception ? Or can you give --pf-c-about-modal-box__content--Margin its own value?

File: src/patternfly/components/AboutModalBox/about-modal-box.scss [hunk 6]
```diff
@@ -10,79 +10,72 @@
...
+  --pf-c-about-modal-box__close--lg--PaddingRight: var(--pf-c-about-modal-box--Padding);
   --pf-c-about-modal-box__close--ZIndex: var(--pf-global--ZIndex--2xl);
 
   // Brand img
   --pf-c-about-modal-box__brand-image--Height: #{pf-size-prem(40px)};
 
   // Header
   // Set default padding right, bottom, left to shared values
-  --pf-c-about-modal-box__header--PaddingRight: var(--pf-c-about-modal-box--PaddingRight);
+  --pf-c-about-modal-box__header--PaddingRight: var(--pf-c-about-modal-box--Padding);
```
Comment (christiemolloy) [inline src/patternfly/components/AboutModalBox/about-modal-box.scss]: Potentially combine Padding Right and Padding Left as PaddingX

File: src/patternfly/components/AboutModalBox/about-modal-box.scss [hunk 7]
```diff
@@ -10,79 +10,72 @@
...
   --pf-c-about-modal-box__brand--PaddingTop: var(--pf-global--spacer--2xl);
-  --pf-c-about-modal-box__brand--PaddingRight: var(--pf-c-about-modal-box--PaddingRight);
-  --pf-c-about-modal-box__brand--PaddingBottom: var(--pf-c-about-modal-box--PaddingBottom);
-  --pf-c-about-modal-box__brand--PaddingLeft: var(--pf-c-about-modal-box--PaddingLeft);
+  --pf-c-about-modal-box__brand--PaddingRight: var(--pf-c-about-modal-box--Padding);
+  --pf-c-about-modal-box__brand--PaddingBottom: var(--pf-c-about-modal-box--Padding);
+  --pf-c-about-modal-box__brand--PaddingLeft: var(--pf-c-about-modal-box--Padding);
 
   // Close
   // Set default padding top, right, bottom to shared values
```
Comment (christiemolloy) [inline src/patternfly/components/AboutModalBox/about-modal-box.scss]: IMO `--pf-c-about-modal-box__close--PaddingTop` being assigned to `var(--pf-c-about-modal-box__brand--PaddingTop)` is confusing . I understand that its good to reuse variables but I wonder what happens if the brand PaddingTop changes, and we don't want the box_close PaddingTop to change? But thats just my thought!

File: src/patternfly/components/AboutModalBox/about-modal-box.scss [hunk 8]
```diff
@@ -10,79 +10,72 @@
...
 
   // Element variables
 
   // Brand
   // Set default padding top, right, left to shared values
   --pf-c-about-modal-box__brand--PaddingTop: var(--pf-global--spacer--2xl);
-  --pf-c-about-modal-box__brand--PaddingRight: var(--pf-c-about-modal-box--PaddingRight);
-  --pf-c-about-modal-box__brand--PaddingBottom: var(--pf-c-about-modal-box--PaddingBottom);
-  --pf-c-about-modal-box__brand--PaddingLeft: var(--pf-c-about-modal-box--PaddingLeft);
+  --pf-c-about-modal-box__brand--PaddingRight: var(--pf-c-about-modal-box--Padding);
```
Comment (christiemolloy) [inline src/patternfly/components/AboutModalBox/about-modal-box.scss]: Potentially combine PaddingRight and PaddingLeft as PaddingX

File: src/patternfly/components/AboutModalBox/about-modal-box.scss [hunk 9]
```diff
@@ -10,79 +10,72 @@
...
+  --pf-c-about-modal-box__brand--PaddingBottom: var(--pf-c-about-modal-box--Padding);
+  --pf-c-about-modal-box__brand--PaddingLeft: var(--pf-c-about-modal-box--Padding);
 
   // Close
   // Set default padding top, right, bottom to shared values
   --pf-c-about-modal-box__close--PaddingTop: var(--pf-c-about-modal-box__brand--PaddingTop);
```
Comment (christiemolloy) [inline src/patternfly/components/AboutModalBox/about-modal-box.scss]: IMO --pf-c-about-modal-box__close--PaddingTop being assigned to var(--pf-c-about-modal-box__brand--PaddingTop) is confusing . I understand that its good to reuse variables but I wonder what happens if the brand PaddingTop changes, and we don't want the box_close PaddingTop to change? But thats just my thought!

File: src/patternfly/components/AboutModalBox/about-modal-box.scss (hunk 1 above)
Comment (mcoker) [reply src/patternfly/components/AboutModalBox/about-modal-box.scss]: I agree it's a lot, but the core stuff in button we want to retain (no border, user-select: none). I think this is pushing it, but I'd say the basic function of the button is the same, just a different presentation. @jgiardino what do you think? does this share the essence of the `.pf-c-button` component?

File: src/patternfly/components/AboutModalBox/about-modal-box.scss (hunk 2 above)
Comment (mcoker) [reply src/patternfly/components/AboutModalBox/about-modal-box.scss]: 👍

File: src/patternfly/components/AboutModalBox/about-modal-box.scss (hunk 3 above)
Comment (mcoker) [reply src/patternfly/components/AboutModalBox/about-modal-box.scss]: first bullet 👍

not sure what you mean by the second one though?

Comment (mcoker) [reply src/patternfly/components/AboutModalBox/about-modal-box.scss]: sass doesn't like that :( I opened an issue though, looks like it's fixed in later versions of sass. https://github.com/patternfly/patternfly-next/issues/1033

File: src/patternfly/components/AboutModalBox/about-modal-box.scss (hunk 9 above)
Comment (mcoker) [reply src/patternfly/components/AboutModalBox/about-modal-box.scss]: Good point. This component shares a lot of variables and I think it's OK. We want to assume alignment between a lot of sections of this page, so it makes sense just to share variables. That way we have individual variables we can overwrite (like you can jsut overwrite `--pf-c-about-modal-box__close--PaddingTop` if you change `--pf-c-about-modal-box__brand--PaddingTop` and don't want the close button to move)

**Comment by mcoker** [comment]
Thanks for the awesome review, @christiemolloy!! Made some updates if you want to take another look.

File: src/patternfly/components/AboutModalBox/about-modal-box.scss (hunk 9 above)
Comment (christiemolloy) [reply src/patternfly/components/AboutModalBox/about-modal-box.scss]: Thanks for explaining, happy with that!

File: src/patternfly/components/AboutModalBox/about-modal-box.scss [hunk 10]
```diff
@@ -10,104 +10,93 @@
...
   --pf-c-about-modal-box__header-strapline--PaddingBottom: var(--pf-global--spacer--md);
 
   // Content
   // Using global padding values for margins to prevent scrollbar extending past content
   // Set default margin top, right, bottom, left to shared values
-  --pf-c-about-modal-box__content--MarginTop: var(--pf-c-about-modal-box--PaddingTop);
-  --pf-c-about-modal-box__content--MarginRight: var(--pf-c-about-modal-box--PaddingRight);
-  --pf-c-about-modal-box__content--MarginBottom: var(--pf-c-about-modal-box--PaddingBottom);
-  --pf-c-about-modal-box__content--MarginLeft: var(--pf-c-about-modal-box--PaddingLeft);
+  --pf-c-about-modal-box__content--Margin: var(--pf-c-about-modal-box--Padding);
```
Comment (christiemolloy) [inline src/patternfly/components/AboutModalBox/about-modal-box.scss]: Break margin into MarginX and MarginY?

File: src/patternfly/components/AboutModalBox/about-modal-box.scss (hunk 3 above)
Comment (christiemolloy) [reply src/patternfly/components/AboutModalBox/about-modal-box.scss]: interesting! thanks for creating the issue

File: src/patternfly/components/AboutModalBox/about-modal-box.scss [hunk 11]
```diff
@@ -184,10 +173,7 @@
 .pf-c-about-modal-box__content {
   grid-area: content;
   min-height: var(--pf-c-about-modal-box__content--MinHeight);
-  margin-top: var(--pf-c-about-modal-box__content--MarginTop); // Using margin instead of padding to prevent scrollbar extending to top of content block
-  margin-right: var(--pf-c-about-modal-box__content--MarginRight);
-  margin-bottom: var(--pf-c-about-modal-box__content--MarginBottom);
-  margin-left: var(--pf-c-about-modal-box__content--MarginLeft);
+  margin: var(--pf-c-about-modal-box__content--Margin); // Using margin instead of padding to prevent scrollbar extending to top of content block
```
Comment (christiemolloy) [inline src/patternfly/components/AboutModalBox/about-modal-box.scss]: margin: marginY marginX ?

**Comment by christiemolloy** [comment]
@mcoker left some more comments :)

**Comment by mcoker** [comment]
@christiemolloy thanks! upated.

File: src/patternfly/components/AboutModalBox/about-modal-box.scss [hunk 12]
```diff
@@ -235,33 +225,36 @@
   padding-bottom: var(--pf-c-about-modal-box__close--PaddingBottom);
 
   @media only screen and (min-width: $pf-global--breakpoint--sm) {
+    --pf-c-about-modal-box__close--PaddingRight: var(--pf-c-about-modal-box__close--sm--PaddingRight);
```
Comment (mattnolting) [inline src/patternfly/components/AboutModalBox/about-modal-box.scss]: Should this be in the top of file?

File: src/patternfly/components/AboutModalBox/about-modal-box.scss [hunk 13]
```diff
@@ -235,33 +225,36 @@
...
+
     grid-area: 1 / 2;
     justify-content: center; // center the close over the narrow hero image
-    padding-right: 0;
   }
 
   @media only screen and (min-width: $pf-global--breakpoint--lg) {
+    --pf-c-about-modal-box__close--PaddingRight: var(--pf-c-about-modal-box__close--lg--PaddingRight);
```
Comment (mattnolting) [inline src/patternfly/components/AboutModalBox/about-modal-box.scss]: Should this be in the top of file?

Comment (mcoker) [reply src/patternfly/components/AboutModalBox/about-modal-box.scss]: yepper

File: src/patternfly/components/AboutModalBox/about-modal-box.scss (hunk 12 above)
Comment (mcoker) [reply src/patternfly/components/AboutModalBox/about-modal-box.scss]: mmmhmmm

**Comment by mcoker** [comment]
thanks @mattnolting, updated the branch and also put in a fix for https://github.com/patternfly/patternfly-next/issues/1034

**Comment by matthewcarleton** [comment]
closing this. @mcoker is going to open a new PR to make this simpler to merge :)

**Pull request closed without merging**